src/chainlit/message.py

```diff
@@ -153,10 +154,18 @@ def __init__(
         actions: Optional[List[Action]] = None,
         elements: Optional[List[ElementBased]] = None,
     ):
-        self.content = content
+        self.language = language
+
+        if type(content) is dict:
+            self.content = json.dumps(content, indent=4)
+            self.language = "json"
+        elif type(content) is str:
```

Comment:
let's allow subclasses as well

```suggestion
        elif isinstance(content, str):
```